htdocs/core/class/cemailtemplate.class.php

```diff
@@ -0,0 +1,582 @@
+
+		if ($result > 0 && !empty($this->table_element_line) && empty($nolines)) {
+			$this->fetchLines($noextrafields);
+		}
+		return $result;
+	}
+
+	/**
+	 *	Get email template from database.
+	 *
+	 *	@param      int			$id       	row Id of email template
+	 *	@param      string		$label    	label of email template
+	 *	@return     int         			>0 if OK, <0 if KO, 0 if not found
+	 */
+	public function apifetch($id, $label = '')
+	{
+		global $db, $conf;
+
+		// Check parameters
+		if (empty($id) && empty($label)) {
+			return -1;
+		}
+
+		$sql = "SELECT e.rowid, e.entity, e.module, e.type_template, e.lang,";
+		$sql .= " e.private, e.fk_user, e.datec, e.tms, e.label, e.position,";
+		$sql .= " e.defaultfortype, e.enabled, e.active, e.email_from, e.email_to,";
+		$sql .= " e.email_tocc, e.email_tobcc, e.topic, e.joinfiles, e.content,";
+		$sql .= " e.content_lines FROM ".$this->db->prefix().$this->table_element." as e";
+		if ($id) {
+			$sql .= " WHERE e.rowid = ".((int) $id);
+		}
+		if ($label) {
+			$sql .= " WHERE e.label = '".$this->db->escape($label)."'";
+		}
+		$sql .= " AND e.entity = ".((int) $this->entity);
+
+		dol_syslog(get_class($this)."::fetch", LOG_DEBUG);
+		$result = $this->db->query($sql);
+		if ($result) {
+			$obj = $this->db->fetch_object($result);
+			if ($obj) {
+				$this->id = (int) $obj->rowid;
+				$this->entity = (int) $obj->entity;
+
+				$this->active = (int) $obj->active;
+				$this->content = (string) $obj->content;
+				$this->content_lines = (string) $obj->content_lines;
+				$this->datec = (int) $obj->datec;
+				$this->defaultfortype = (int) $obj->defaultfortype;
+				$this->email_from = (string) $obj->email_from;
+				$this->email_to = (string) $obj->email_to;
+				$this->email_tobcc = (string) $obj->email_tobcc;
+				$this->email_tocc = (string) $obj->email_tocc;
+				$this->enabled = (string) $obj->enabled;
+				$this->fk_user = (int) $obj->fk_user;
+				$this->joinfiles = (int) $obj->joinfiles;
+				$this->label = (string) $obj->label;
+				$this->lang = (string) $obj->lang;
+				$this->module = (string) $obj->module;
+				$this->position = (int) $obj->position;
+				$this->private = (int) $obj->private;
+				$this->tms = $obj->tms;
+				$this->topic = (string) $obj->topic;
+				$this->type_template = (string) $obj->type_template;
+
+				// direct copy from facture.class.php
+				$this->date_creation = $this->db->jdate($obj->datec);
+
+				return 1;
+			} else {
+				if ($id) {
+					$this->error = 'Email template with id '.((string) $id).' not found sql='.$sql;
+				}
+				if ($label) {
+					$this->error = 'Email template with label '.$label.' not found sql='.$sql;
+				}
+				return 0;
+			}
+		} else {
+			$this->error = $this->db->error();
+			return -1;
+		}
+	}
+}
+
+/**
+ * old class name for Object of table llx_c_email_templates
+ * I prefer the cEmailTemplate name as it better reflects the database
+ */
+class ModelMail extends cEmailTemplate
```

Comment:
@eldy  I think this PR is ready. At what point can we deprecate the old name `ModelMail` and remove this class definition?

It does not seem to be used in ANY of the files in this github repo, but maybe someone uses it in a module somewhere?